src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -2246,28 +2225,15 @@ splitOnSuffixSeq withSep patArr (Fold fstep initial done) (Stream step state) =
             case res of
                 Yield x s -> do
                     let wrd1 = addToWord wrd x
-                    if withSep
-                    then do
-                        fs1 <- fstep fs x
-                        if idx == maxIndex
-                        then do
-                            if wrd1 .&. wordMask == wordPat
-                            then do
-                                r <- done fs
-                                let next = SplitOnSuffixSeqWordInit s
-                                skip $ SplitOnSuffixSeqYield r next
-                            else skip $ SplitOnSuffixSeqWordLoop wrd1 s fs1
-                        else go SPEC (idx + 1) wrd1 s fs1
+                    fs1 <- if withSep then fstep fs x else return fs
```

Comment:
Can you change this back to normal, I've written it like such with partial folds in mind.

Reply:
I do not see how partial folds will change things fundamentally. You could just change `return fs` to `return $ Partial 0 fs` and the rest all should be handled in the same way. Am I missing something?

Reply:
Yeah, I want to write code in a very idiomatic and straightforward way. Once it has the same performance, I'll abstract things out bit by bit.
The last time I tried to abstract this out, the benchmarks were not the prettiest. I made a few other changes then, so technically cannot blame it on this. I'll reduce duplication as much as possible in the end.

Reply:
I will suggest that you can change it to whatever form when you make those changes. At this time we should write whatever is best now.